packages/core/src/di/reflective_key.ts

```diff
@@ -35,7 +37,8 @@ export class ReflectiveKey {
    */
   constructor(public token: Object, public id: number) {
     if (!token) {
-      throw new Error('Token must be defined!');
+      throw new RuntimeError(
+          RuntimeErrorCode.INJECTION_TOKEN_UNDEFINED, ngDevMode && 'Token must be defined!');
```

Comment:
I think we can name this error code `MISSING_INJECTION_TOKEN`, similar to an existing `MISSING_INJECTION_CONTEXT` token:
```suggestion
          RuntimeErrorCode.MISSING_INJECTION_TOKEN, ngDevMode && 'Token must be defined!');
```

Reply:
@AndrewKushnir, I've changed the error code to `MISSING_INJECTION_TOKEN` so it's more aligned with the existing ones.